dashboard/app/views/home/_survey_diversity.haml

```diff
@@ -1,91 +1,82 @@
-              - else
-                - text = "Other"
-              .dot{style: "margin-top:10px"}
-                %input{style: "margin-right:3px; vertical-align:top", id: "radio_#{index}", type: "radio", name: "survey[survey2016_foodstamps]", value: index, onclick: "enableSurveySubmit()" }
-                %label{style: "display: inline; margin-top:2px; font-size: 13px; padding-right:20px", for: "radio_#{index}"}= text
+    = form_for(SurveyResult.new, html: {id: 'survey', onsubmit: "return window.submitSurvey(true)"}) do |f|
+      .survey_left
+        %ol
+          %li
+            Of your students using Code.org's courses, how many are
+        .dots
+          - SurveyResult::ETHNICITIES.each do |key, value|
+            %input#ethnicity_input{maxlength: "3", name: "survey[survey2017_ethnicity_#{key}]", placeholder: "0"}
+            = value
+            %br/
+      .survey_right
 
-        #submit{style: "text-align: right"}
-          %button.btn-primary#submitSurvey{type: "submit", disabled: true} Submit
-          %span{onclick: "return submitSurvey(false)", style: "cursor: pointer"} No thanks
-  %br/
+        %ol{start: 2}
+          %li
+            What is the percentage of students with free and reduced meals (FARMs) at your school?
+            .subtext
+              Only answer this if you're using Code Studio during school hours.
+        .dots
+          - (0..9).each do |index|
+            - if index.between?(0,7)
+              - value = index * 10
+              - max_value = value == 70 ? 100 : (value+9)
+              - text = "#{value}-#{max_value}%"
+            - elsif index == 8
+              - text = "My school has a Title I designation, but I’m not sure what our rate of Free and Reduced lunch is."
+            - else
+              - text = "Other"
+            .dot
+              %input{id: "radio_#{index}", type: "radio", name: "survey[survey2017_foodstamps]", value: index}
+              %label{for: "radio_#{index}"}= text
 
-:javascript
+      %input{type: "hidden", name: "survey[kind]", value: SurveyResult::DIVERSITY_2017}
 
-  $.ready(function() {
-    // placeholder text for non-HTML5 browsers
-    $('input[type=text]').placeholder();
-  });
+      #submit
+        %button.btn-primary#submitSurvey{type: "submit"} Submit
+        %span#noanswer{onclick: "return submitSurvey(false)"} Prefer not to answer
+  %br/
 
-  function enableSurveySubmit()
-  {
-    $("#submitSurvey").prop("disabled",false);
-  }
+:javascript
 
   function submitSurvey(submittedValue)
   {
     // make sure each ethnicity is a positive integer
-    $(".ethnicity_input").each(function(index) {
+    $("#ethnicity_input").each(function(index) {
       $(this).val($(this).val().replace(/[^0-9]/g, ''));
     });
 
     if (submittedValue)
     {
       $("#surveybody").hide();
-      $("#surveythanks").show();
+      $("#surveythanks").fadeIn();
     }
     else
     {
       $("#survey")[0].reset();
```

Comment:
This just resets the survey form.  The `form_for` line specifies an id of `survey` for the form.